### PR TITLE
Exposing some Channel parameters

### PR DESCRIPTION
**Pull request by LucasVanDongen**
I want to be able to see what topic a Channel has from code, so I exposed it as well as a few other parameters I might want to be able to see.

### CODE REVIEW

File: Sources/Constants.swift [hunk 1]
```diff
@@ -10,7 +10,7 @@
 public let PHOENIX_DEFAULT_TIMEOUT: Int = 10000
 
 /// Default heartbeat interval set to 30 seconds
-public let PHOENIX_DEFAULT_HEARTBEAT: Int = 30000
+public let PHOENIX_DEFAULT_HEARTBEAT: Int = 5000
```
Comment (dsrees) [inline Sources/Constants.swift]: Why was this changed?

File: Sources/Channel.swift [hunk 2]
```diff
@@ -27,18 +27,18 @@ import Swift
...
     /// The topic of the Channel. e.g. "rooms:friends"
-    let topic: String
+    public let topic: String
     
     /// The params sent when joining the channel
     var params: Payload
 
     /// The Socket that the channel belongs to
-    weak var socket: Socket?
+    public weak var socket: Socket?
```
Comment (dsrees) [inline Sources/Channel.swift]: I don't like exposing the socket outside of the channel, especially since it is a `weak var` now and not an immutable `let` Also, you should already have access to the `socket` since it is required to create the channel in the first place

File: Sources/Constants.swift (hunk 1 above)
Comment (LucasVanDongen) [reply Sources/Constants.swift]: I have no idea, must've slipped through the cracks as I didn't notice it while pushing. I branched from your master before pushing some changes. I think our history got a bit mixed up.

This could be made variable with a 30 second default though.

File: Sources/Channel.swift [hunk 3]
```diff
@@ -27,18 +27,18 @@ import Swift
...
 
     
     
     /// Current state of the Channel
-    var state: ChannelState
+    private(set) public var state: ChannelState
```
Comment (dsrees) [inline Sources/Channel.swift]: This breaks some tests. Also, you can access the state of the channel already by calling `isClosed`, `isJoined`, etc. See [more here](https://github.com/davidstump/SwiftPhoenixClient/blob/796d7fa7a08b9fd781bc712f2df423d4b2c73dfb/Sources/Channel.swift#L357). So I don't think this property needs to be exposed

File: Sources/Channel.swift (hunk 2 above)
Comment (LucasVanDongen) [reply Sources/Channel.swift]: I have a piece of code that I'm not sure how to rebuild otherwise:

```            guard !type(of: self).connectors.contains(where: { (connector) -> Bool in
                connector.channel?.topic == topic
            }) else {
                return
            }```

I built it before reconnection logic really worked though.

Could it be `private(set)` so it would be effectively read-only?

File: Sources/Channel.swift (hunk 3 above)
Comment (LucasVanDongen) [reply Sources/Channel.swift]: OK I can live without this

File: Sources/Channel.swift (hunk 2 above)
Comment (dsrees) [reply Sources/Channel.swift]: It looks like you're accessing only the channel's `topic` in that piece of code, which I am fine with exposing. Am I reading that correctly?

File: Sources/Constants.swift (hunk 1 above)
Comment (dsrees) [reply Sources/Constants.swift]: The socket heartbeat can be configured by setting it on the socket
```
socket.heartbeatIntervalMs = 5000
```

This is a constant to be used as the preset default. 
```
/// Interval between sending a heartbeat
public var heartbeatIntervalMs: Int = PHOENIX_DEFAULT_HEARTBEAT
```

File: Sources/Channel.swift (hunk 2 above)
Comment (LucasVanDongen) [reply Sources/Channel.swift]: That's right. An alternative way to access it would be great

File: Sources/Constants.swift (hunk 1 above)
Comment (LucasVanDongen) [reply Sources/Constants.swift]: OK problem solved then

File: Sources/Channel.swift (hunk 2 above)
Comment (dsrees) [reply Sources/Channel.swift]: I'm fine exposing `topic` as a `public let topic`, however you are also exposing `socket` as `public socket`, which I don't think should be allowed and it looks like from your code snippet above, you're not making use of `socket` anyways

Comment (LucasVanDongen) [reply Sources/Channel.swift]: so in `Channel.swift` we would add something like:

```let topic: String```

which gets set in the constructor?

Do you want me to create a new branch and pull request?

Comment (dsrees) [reply Sources/Channel.swift]: You've already made the change `public let topic: String` in this PR [at line 30](https://github.com/davidstump/SwiftPhoenixClient/pull/115/files#diff-0f6463bf4fb25f9638a8ad7fca2193ffR29). which I think is 👍 which makes `channel.topic` accessible

I would just like to see the `public` modifier removed from the socket declaration [at line 36](https://github.com/davidstump/SwiftPhoenixClient/pull/115/files#diff-0f6463bf4fb25f9638a8ad7fca2193ffR36) so that `channel.socket` is not accessible

**Comment by dsrees** [comment]
I've made the changes in #115 and will be pushing out a release
